odoo/models.py

```diff
@@ -3397,6 +3411,12 @@ def unlink(self):
         self.check_access_rights('unlink')
         self._check_concurrency()
 
+        from odoo.addons.base.models.ir_model import MODULE_UNINSTALL_FLAG
+        for func in self._ondelete:
+            if self._context.get(MODULE_UNINSTALL_FLAG) and not func._ondelete:
```

Comment:
a bit confusing to read without knowing that `func._ondelete = at_uninstall`

Reply:
Yes maybe `_ondelete` should not be a boolean but an enum, or at least a pair of integer constants?

Reply:
I decided to go with Raphael's [suggestion](https://github.com/odoo/odoo/pull/58517#discussion_r509980103) and simply add a comment, I agree it's not great but it's consistent with the other decorators (`attrsetter` is not the easiest pill to swallow though)